launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FileUploadServlet.java

```diff
@@ -0,0 +1,291 @@
+  private String makeMp3Link(String entityId) {
+    String link = "https://storage.googleapis.com/" + BUCKET_NAME + "/" + entityId;
+    return link;
+  }
+
+  /**
+  * Helper method for repeated code in catching exceptions.
+  */
+  private void writeResponse(HttpServletResponse res, String message, int statusCode) throws IOException {
+    res.setContentType("text/html");
+    res.getWriter().println(message);
+    res.setStatus(statusCode);
+  }
+
+  /**
+   * Requests user inputs in form fields, then creates Entity and places in Datastore.
+   * @throws ServletException
+   */
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws IllegalArgumentException, IOException {
+    String podcastTitle = req.getParameter(PODCAST_TITLE);
+    String description = req.getParameter(DESCRIPTION);
+    String language = req.getParameter(LANGUAGE);
+    String email = req.getParameter(EMAIL);
+
+    // TO-DO after merging: add this in and remove email parameter (because LoginServlet will be set up)
+    // UserService userService = UserServiceFactory.getUserService();
+    // if (userService.isUserLoggedIn()) {
+    //   email = userService.getCurrentUser().getEmail();
+    // }
+
+    // TO-DO after merging: move validation to common place 
+    if (Strings.isNullOrEmpty(podcastTitle)) {
+      throw new IllegalArgumentException("No Title inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(description)) {
+      throw new IllegalArgumentException("No description inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(language)) {
+      throw new IllegalArgumentException("No language inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(email)) {
+      throw new IllegalArgumentException("You are not logged in. Please try again.");
+    }
+
+    // Create entity with all desired attributes
+    Entity userFeedEntity = new Entity(USER_FEED);
+    userFeedEntity.setProperty(PODCAST_TITLE, podcastTitle);
+    userFeedEntity.setProperty(DESCRIPTION, description);
+    userFeedEntity.setProperty(LANGUAGE, language);
+    userFeedEntity.setProperty(EMAIL, email);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key entityKey = datastore.put(userFeedEntity);
+
+    String entityId = KeyFactory.keyToString(entityKey);
+    String mp3Link = makeMp3Link(entityId);
+
+    // Create embedded entity to store MP3 data
+    EmbeddedEntity mp3 = new EmbeddedEntity();
+    mp3.setProperty(ID, entityId);
+    mp3.setProperty(MP3_LINK, mp3Link);
+    mp3.setProperty(EMAIL, email);
+
+    // Retrieve entity from Datastore
+    Entity savedEntity;
+    try {
+      savedEntity = datastore.get(entityKey);
+    } catch(EntityNotFoundException e) {
+      throw new IOException("Entity not found in Datastore.");
+    }
+
+    // Generate xml string
+    // String name, String email, String podcastTitle, String description, String category, String language
```

Comment:
we can remove this comment. method below has variables named properly so it's already clear enough